galasa-extensions-parent/dev.galasa.auth.couchdb/src/main/java/dev/galasa/auth/couchdb/internal/CouchdbAuthStoreValidator.java

```diff
@@ -9,23 +9,198 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
 import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+
+import com.google.gson.JsonSyntaxException;
 
 import dev.galasa.extensions.common.couchdb.CouchdbBaseValidator;
+import dev.galasa.extensions.common.couchdb.CouchdbClashingUpdateException;
 import dev.galasa.extensions.common.couchdb.CouchdbException;
+import dev.galasa.auth.couchdb.internal.beans.*;
 import dev.galasa.extensions.common.api.HttpRequestFactory;
+import dev.galasa.framework.spi.utils.GalasaGson;
+import dev.galasa.framework.spi.utils.ITimeService;
+import dev.galasa.framework.spi.utils.SystemTimeService;
+
+import static dev.galasa.auth.couchdb.internal.Errors.*;
 
 public class CouchdbAuthStoreValidator extends CouchdbBaseValidator {
 
     private final Log logger = LogFactory.getLog(getClass());
+    private final GalasaGson gson = new GalasaGson();
+
+
+    public static final String DB_TABLE_TOKENS_DESIGN = "function (doc) { if (doc.owner && doc.owner.loginId) {emit(doc.owner.loginId, doc); } }";
```

Comment:
A comment as to what this function is doing would be good - saves others from having to research things